library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/api/reloader/SimpleResourceReloader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api.reloader;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.resource.ResourceReloader;
+import net.minecraft.resource.SynchronousResourceReloader;
+import net.minecraft.util.profiler.Profiler;
+
+/**
+ * A simplified version of the "resource reload listener" interface, hiding the
+ * peculiarities of the API.
+ *
+ * <p>In essence, there are two stages:
+ *
+ * <ul>
+ *     <li>{@linkplain #load(ResourceManager, Profiler, Executor)}: create an instance of your data object
+ * containing all loaded and processed information,
+ *     <li>{@linkplain #apply(Object, ResourceManager, Profiler, Executor)}: apply the information from the data object
+ * to the game instance.
+ * </ul>
+ *
+ * <p>The load stage should be self-contained as it can run on any thread! However,
+ * the apply stage is guaranteed to run on the game thread.
+ *
+ * <p>For a fully synchronous alternative, consider using
+ * {@link SynchronousResourceReloader} in conjunction with
+ * {@link IdentifiableResourceReloader}.
+ *
+ * @param <T> The data object.
```

Comment:
Might be more accurate to describe T as the type of the intermediate reloading state?

Reply:
Hopefully the new doc is better?